presto-docs/src/main/sphinx/admin/resource-groups.rst

```diff
@@ -9,287 +9,436 @@ Except for the limit on queued queries, when a resource group runs out of a reso
+
+Sub-Groups and Query Acceptance
+-------------------------------
+
+Sub-groups allow for hierarchical resource allocation, where each sub-group can have its own resource limits and
+queueing policies.A resource group that accepts queries directly is a leaf group, and it executes queries using
+its allocated resources.
+
+The resource groups and associated selection rules are configured by a manager, which is pluggable.
+Presto resource management can be done in two ways:
+
+* File-Based Resource Management
+* Database-Based Resource Management
+
+File-Based Resource Management
+------------------------------
+
+In a file-based resource manager, configuration information about the resource groups is stored in a JSON file.
+This file contains definitions of all resource groups and the rules to select them.
+The configuration file is loaded and used at the start of the Presto server.
+Any changes to the file require a restart of the Presto server to take effect.
+
+Database-Based Resource Management
+----------------------------------
+
+In a database-based resource manager, configuration information about the resource groups is stored in a relational
+database. The database contains definitions of all resource groups and the rules to select them.
+Unlike the file-based resource manager, changes to the configuration in the database take effect
+immediately and do not require a restart of the Presto server.
+
+Both methods have their pros and cons. File-based resource management is simpler to set up but less flexible,
+while database-based resource management is more complex to set up but offers more flexibility and dynamic changes.
+
+File Resource Group Manager
+---------------------------
+
+The File Resource Group Manager in PrestoDB is a way to manage resources using a JSON configuration file.
+This file contains definitions of all resource groups and rules for selecting the appropriate resource group
+for a given query.
+
+To set up a File-based Resource Group Manager:
+
+1. Create the file ``etc/resource-groups.properties``.
+
+2. Set the ``resource-groups.configuration-manager`` property to ``file``.
+
+3. Create a JSON file: This file should contain the definitions for your resource groups.
+   Each resource group can specify things like the maximum memory, the maximum queued queries,
+   and the maximum running queries.
+
+4. Specify the file in the Presto configuration: In the ``etc`` directory of your Presto installation,
+   create a file named ``resource-groups.properties``. In this file, add a line that specifies the location of
+   your JSON file: ``resource-groups.config-file = <file_path>``, where ``<file_path>`` is the path to your JSON file.
+
+5. Restart the Presto server: After creating the JSON file and pointing to it in the Presto configuration,
+   restart your Presto server. The new resource groups will take effect after the restart.
+
```

Comment:
Suggest moving this code block example of ``etc/resource-groups.properties`` into step 4, with an introductory "Your ``etc/resource-groups.properties`` should now be similar to this following example:" - however you want to explain this code block. 

I suggest moving it to before step 5 Restart the Presto server because the user should have their file be similar to this example before the restart, yes?